codebundles/k8s-pvc-healthcheck/runbook.robot

```diff
@@ -0,0 +1,137 @@
+    ${kubeconfig}=    RW.Core.Import Secret
+    ...    kubeconfig
+    ...    type=string
+    ...    description=The kubernetes kubeconfig yaml containing connection configuration used to connect to cluster(s).
+    ...    pattern=\w*
+    ...    example=For examples, start here https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/
+    ${kubectl}=    RW.Core.Import Service    kubectl
+    ...    description=The location service used to interpret shell commands.
+    ...    default=kubectl-service.shared
+    ...    example=kubectl-service.shared
+    ${KUBERNETES_DISTRIBUTION_BINARY}=    RW.Core.Import User Variable    KUBERNETES_DISTRIBUTION_BINARY
+    ...    type=string
+    ...    description=Which binary to use for Kubernetes CLI commands.
+    ...    enum=[kubectl,oc]
+    ...    example=kubectl
+    ...    default=kubectl
+    ${CONTEXT}=    RW.Core.Import User Variable    CONTEXT
+    ...    type=string
+    ...    description=Which Kubernetes context to operate within.
+    ...    pattern=\w*
+    ...    example=my-main-cluster
+    ${NAMESPACE}=    RW.Core.Import User Variable    NAMESPACE
+    ...    type=string
+    ...    description=The name of the namespace to search.   
+    ...    pattern=\w*
+    ...    example=otel-demo
+    ...    default=''
+    Set Suite Variable    ${kubeconfig}    ${kubeconfig}
+    Set Suite Variable    ${KUBERNETES_DISTRIBUTION_BINARY}    ${KUBERNETES_DISTRIBUTION_BINARY}
+    Set Suite Variable    ${CONTEXT}    ${CONTEXT}
+    Set Suite Variable    ${NAMESPACE}    ${NAMESPACE}
+    Set Suite Variable    ${env}    {"KUBECONFIG":"./${kubeconfig.key}"}
+
+*** Tasks ***
+Fetch Events for Unhealthy Kubernetes Persistent Volume Claims
+    [Documentation]    Lists events related to persistent volume claims within the desired namespace that are not bound to a persistent volume.
+    [Tags]    PVC    List    Kubernetes    Storage    PersistentVolumeClaim    Events
+    ${unbound_pvc_events}=    RW.CLI.Run Cli
+    ...    cmd=for pvc in $(${KUBERNETES_DISTRIBUTION_BINARY} get pvc -n ${NAMESPACE} -o json | jq -r '.items[] | select(.status.phase != "Bound") | .metadata.name'); do ${KUBERNETES_DISTRIBUTION_BINARY} get events -n ${NAMESPACE} --field-selector involvedObject.name=$pvc -o json | jq '.items[]| "Last Timestamp: " + .lastTimestamp + " Name: " + .involvedObject.name + " Message: " + .message'; done
+    ...    target_service=${kubectl}
+    ...    env=${env}
+    ...    secret_file__kubeconfig=${kubeconfig}
+    ...    render_in_commandlist=true
+    ${regexp}=    Catenate
+    ...    (?m)(?P<line>.+)
+    RW.CLI.Parse Cli Output By Line
+    ...    rsp=${unbound_pvc_events}
+    ...    lines_like_regexp=${regexp}
+    ...    set_severity_level=1
+    ...    set_issue_expected=PVCs should be bound
+    ...    set_issue_actual=PVCs found pending with the following events
+    ...    set_issue_title=PVC Errors & Events
+    ...    line__raise_issue_if_contains=Name
```

Comment:
Did `_line__raise_issue_if` not work for you? the `_line` should be predefined for you. I see you're making a `line` capture group here.